### PR TITLE
mounter: fix auto_mount to verify versions

### PR DESCRIPTION
**Pull request by netanelc305**

### CODE REVIEW

File: pymobiledevice3/cli/mounter.py [hunk 1]
```diff
@@ -157,17 +156,31 @@ def mounter_auto_mount(lockdown: LockdownClient, xcode: str, version: str):
...
-                f'Please make sure your user has the necessary permissions')
-            return
-
     image_path = Path(image_path)
     signature = Path(signature)
+
+    available_versions = get_all_versions()
+
+    if version not in available_versions:
+        logger.error(f"Unable to find DeveloperDiskImage for {version}. available versions: {available_versions} ")
```
Comment (doronz88) [inline pymobiledevice3/cli/mounter.py]: - change to single quote
- remove trailing space

File: pymobiledevice3/cli/mounter.py [hunk 2]
```diff
@@ -157,17 +156,31 @@ def mounter_auto_mount(lockdown: LockdownClient, xcode: str, version: str):
...
+        return
+
+    try:
+        if not developer_disk_image_dir.exists():
```
Comment (doronz88) [inline pymobiledevice3/cli/mounter.py]: use `mkdir(exists=True)`

**Review by doronz88 — CHANGES_REQUESTED** [review]

File: pymobiledevice3/cli/mounter.py (hunk 1 above)
Comment (netanelc305) [reply pymobiledevice3/cli/mounter.py]: k

File: pymobiledevice3/cli/mounter.py (hunk 2 above)
Comment (netanelc305) [reply pymobiledevice3/cli/mounter.py]: k

**Review by doronz88 — APPROVED** [review]

**Pull request merged**